src/domain/orders/details/returns/index.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState, useEffect, useContext } from "react"
+import Modal from "../../../../components/molecules/modal"
+import CurrencyInput from "../../../../components/organisms/currency-input"
+import Button from "../../../../components/fundamentals/button"
+import Select from "../../../../components/molecules/select"
+import Medusa from "../../../../services/api"
+import { filterItems } from "../utils/create-filtering"
+import { getErrorMessage } from "../../../../utils/error-messages"
+import EditIcon from "../../../../components/fundamentals/icons/edit-icon"
+import { displayAmount } from "../../../../utils/prices"
+import InfoTooltip from "../../../../components/molecules/info-tooltip"
+import CheckIcon from "../../../../components/fundamentals/icons/check-icon"
+import Spinner from "../../../../components/atoms/spinner"
+import RMAShippingPrice from "../../../../components/molecules/rma-select-shipping"
+import RMASelectProductTable from "../../../../components/organisms/rma-select-product-table"
+import LayeredModal, {
+  LayeredModalContext,
+} from "../../../../components/molecules/modal/layered-modal"
+
+const removeNullish = (obj) =>
+  Object.entries(obj).reduce((a, [k, v]) => (v ? ((a[k] = v), a) : a), {})
+
+const ReturnMenu = ({ order, onReturn, onDismiss, toaster }) => {
+  const layoutmodalcontext = useContext(LayeredModalContext)
+
+  const [submitting, setSubmitting] = useState(false)
+  const [refundEdited, setRefundEdited] = useState(false)
+  const [refundable, setRefundable] = useState(0)
+  const [refundAmount, setRefundAmount] = useState(0)
+  const [toReturn, setToReturn] = useState({})
+  const [quantities, setQuantities] = useState({})
+  const [useCustomShippingPrice, setUseCustomShippingPrice] = useState(false)
+
+  const [shippingLoading, setShippingLoading] = useState(true)
+  const [shippingOptions, setShippingOptions] = useState([])
+  const [noNotification, setNoNotification] = useState(order.no_notification)
+  const [shippingPrice, setShippingPrice] = useState<number>()
+  const [shippingMethod, setShippingMethod] = useState(null)
+
+  const [allItems, setAllItems] = useState<any[]>([])
+
+  useEffect(() => {
+    if (order) {
+      setAllItems(filterItems(order, false))
+    }
+  }, [order])
+
+  useEffect(() => {
+    Medusa.shippingOptions
+      .list({
+        region_id: order.region_id,
+        is_return: true,
+      })
+      .then(({ data }) => {
+        setShippingOptions(data.shipping_options)
+        setShippingLoading(false)
+      })
```

Comment:
Could we use `useAdminShippingOptions` hook here?

Reply:
not at the moment, this: 
```javascript
const { shipping_options } = useAdminShippingOptions({
    region_id: order.region_id,
    is_return: "true",
  })
```
returns all shipping options for the store

Reply:
Hmm strange. Do you have any idea why @zakariaelas ?

Reply:
@pKorsholm Zak pushed a new canary version of medusa-react the other day that fixes this so try to upgrade the version in your PR should work then :)